### DISCUSSION TITLE
Encaminhar request

### DISCUSSION THREAD

**Discussion by rodrigoferreiramonteiro** [HashLoad/horse#348] [Q&A]
Estou criando uma API que vai receber no body do request videos geralmente bem grandes ( Acima de 500 mega )

Estamos usando um parceiro para gerir o repositorio dos videos , que nosso caso é o PANDA VIDEOS 
https://pandavideo.readme.io/reference/upload-video-1

Por questoes de segurança no client nao existe a informação das chaves de autenticacao com a PANDA, sendo assim, o server ao receber o request deve ENCAMINHAR para a api do panda e efetivamente fazer o upload do video.

Alguma ideia de como receber o request (com o video no body) e encaminhar para a API do panda?

**Comment by rodrigoferreiramonteiro** [comment]
Nenhuma ideia de como executar pessoal?

**Comment by alcantarajoaopaulo** [comment]
Já fiz esse encaminhamento em um Hub de Cobrança, para geração de boletos, em duas aplicações feitas em Delphi e NodeJS. Nunca trabalhei com vídeos.

Não sei como está seu projeto atualmente, mas acredito que o primeiro passo seja viabilizar esse procedimento separadamente, ou seja:
- faça a API que recebe o vídeo do app client e guarde-o (no disco ou num S3 da vida)
- faça um projeto simples, VCL, que realiza uploads de vídeos do seu PC para o Panda (usando RESTClient, Indy, RestRequest4D, etc.)
Feito isso, 90% do trabalho estará feito.

Isso porque prover esse mecânica não tem a ver com o Horse.
O Horse vai te ajudar a criar a API para seu app client. Simples! Porém processar o vídeo, tratar, armazenar e então preparar um Request para a API do Panda é só você, o Delphi e o componente de requisição que você escolher.

No final, você junta os dois: o seu endpoint POST /video feito em Horse acionará uma função em outro local que dispara a requisição para o Panda.

Quanto as credencias: você deverá ter um repositório de dados para identificar o client e então localizar numa tabelas as credenciais da conta Panda que serão usadas no POST do vídeo.

**Reply by rodrigoferreiramonteiro** [reply]
Opa alcantara, 

Eu queria evitar ter que salvar o video, para reenviar no request(post) para o panda, pois é um processo lento

Mandar criar um stream de um video de 4gb demora bastante tempo e NA TEORIA, este video ja esta no body do request que recebi no horse.

Eu queria pegar o body do THorseRequest que ja esta com o video ali e usar ele para adicionar no body do upload para o panda, desta forma nao preciso salvar em disco para reprocessar

Alguma ideia ?
